### ISSUE TITLE
`bufler-list-switch-buffer-action` not respected when using `bufler-switch-buffer`

### ISSUE THREAD

**Issue by Leon-Plickat** [alphapapa/bufler.el#98]
I set `bufler-list-switch-buffer-action` to `'((display-buffer-same-window))`, however when switching buffers with `bufler-switch-buffer` it still acts as if `display-buffer-reuse-window` is being used (other window displaying the selected buffer is focused, while I want to have the currently selected window display the buffer regardless of whether other windows in the current frame already do so). 

According to package, the installed version is `20240312.552`.

I searched a bit in the source but couldn't find the issue, I'll continue to look for it though.

**Comment by alphapapa** [comment]
Display buffer actions are complicated.  Please try to reproduce the problem in a clean Emacs configuration.  If you can, then please provide a step-by-step recipe.

**Comment by Leon-Plickat** [comment]
This can be reproduced in a clean configuration as follows:

Create a directory `repro`, containing `init.el` with the following contents (copied from my config):

```elisp
(use-package package
  :defer t
  :init
  (package-initialize)
  (add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/") t)
  :custom
  (package-native-compile t))

(use-package bufler
  :defer t
  :ensure t
  :custom
  (bufler-indent-per-level 4)
  (bufler-list-switch-buffer-action '((display-buffer-same-window)))

  :bind (("C-x C-b" . bufler)
	 ("C-x b" . bufler-switch-buffer)))
```

Then run `emacs --init-directory=repro/`. Wait until this emacs instance has launched and the packages are installed / compiled. Split via `C-x 2`. Execute `M-x scratch-buffer RET`. The focused window will now display the scratch buffer, the other the welcome splash buffer. Press `C-x b` to call `bufler-switch-buffer` and type `*GN` followed by Tab followed by `RET`. The window showing the splash buffer will be focused, the other window still displays the scratch buffer. The behaviour I would expect is that the focused window remains focused but switches it's buffer to the splash buffer, which is then displayed in both windows.

I hope this is useful enough, let me know if you need anything more.

**Comment by Leon-Plickat** [comment]
![grafik](https://github.com/user-attachments/assets/2f309b96-f442-47a9-a402-6dd3d7db34f2)

Ah, I think this is the responsible code for buffer switching (`bufler-workspace.el`). It does indeed not look at `bufler-list-switch-buffer-action`.

So this turns out to be a feature request rather than a bug report I guess, sorry for the noise.

**Comment by alphapapa** [comment]
> Or does the "list" in `bufler-list-switch-buffer-action` imply it's not meant to change the behaviour of `bufler-switch-buffer`? I could not find a similar variable for that so I assumed this configured both.

That's correct, that option is for the buffer list, not for just any switching command.

> So this turns out to be a feature request rather than a bug report I guess, sorry for the noise.

We could add an option for that, but take a look at the option `switch-to-buffer-obey-display-actions`; maybe it would solve this problem for you.
